src/runtime_src/core/include/xrt/xrt_bo.h

```diff
@@ -117,12 +117,13 @@ class bo
    */
   enum class flags : uint32_t
   {
-    normal      = 0,
-    cacheable   = XRT_BO_FLAGS_CACHEABLE,
-    device_only = XRT_BO_FLAGS_DEV_ONLY,
-    host_only   = XRT_BO_FLAGS_HOST_ONLY,
-    p2p         = XRT_BO_FLAGS_P2P,
-    svm         = XRT_BO_FLAGS_SVM,
+    normal        = 0,
+    cacheable     = XRT_BO_FLAGS_CACHEABLE,
+    device_only   = XRT_BO_FLAGS_DEV_ONLY,
+    host_only     = XRT_BO_FLAGS_HOST_ONLY,
+    p2p           = XRT_BO_FLAGS_P2P,
+    svm           = XRT_BO_FLAGS_SVM,
+    device_memory = XRT_BO_FLAGS_CARVEOUT,
```

Comment:
Since `device_memory` is a public flag, you have to document it along with the other flags above.  Explain what it means, and why it is different from `device_only`.

Reply:
Thanks Soren, for the comment. I added the comment.